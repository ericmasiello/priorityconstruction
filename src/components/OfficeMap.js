import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { pxToRem } from '../styles/utils';

export const OfficeMap = props => {
  const { title, address, width, height, mapKey, ...rest } = props;

  return (
    <iframe
      title={title}
      src={`https://www.google.com/maps/embed/v1/place?q=${address}&key=${mapKey}`}
      {...rest}
    />
  );
};

OfficeMap.propTypes = {
  mapKey: PropTypes.string.isRequired,
  address: PropTypes.string,
  title: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
};

OfficeMap.defaultProps = {
  address: '1315%20West%20Hamburg%20Street%2C%20Baltimore%2C%20MD%2021203',
  title: 'Priority Construction',
};

export default styled(OfficeMap)`
  border: none;
  ${({ width }) => `width: ${width || pxToRem(450)};`};
  ${({ height }) => `height: ${height || pxToRem(350)};`};
`;
