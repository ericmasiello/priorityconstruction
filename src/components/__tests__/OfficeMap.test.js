import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import OfficeMap, { OfficeMap as Raw } from '../OfficeMap';

it('should render', () => {
  const component = renderer.create(<OfficeMap mapKey="foo" />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = shallow(<Raw mapKey="foo">Test</Raw>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<Raw mapKey="foo" data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<Raw mapKey="foo" className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <iframe>', () => {
  const wrapper = shallow(<Raw mapKey="foo" />);

  expect(wrapper.type()).toBe('iframe');
});
