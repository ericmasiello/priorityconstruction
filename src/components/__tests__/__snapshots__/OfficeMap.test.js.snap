// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  border: none;
  width: 28.125rem;
  height: 21.875rem;
}

<iframe
  className="c0"
  src="https://www.google.com/maps/embed/v1/place?q=1315%20West%20Hamburg%20Street%2C%20Baltimore%2C%20MD%2021203&key=foo"
  title="Priority Construction"
/>
`;
